### ISSUE TITLE
camera/multiple-resolution-images_* job is unable to run on the server and core.

### ISSUE THREAD

**Issue by rickwu666666** [canonical/checkbox#650]
### Bug Description

camera/multiple-resolution-images_* job is unable to run on the server and core. This is because it is located in the [client-cert-iot-ubuntucore-20.pxu](https://github.com/canonical/checkbox/blob/main/providers/certification-client/units/client-cert-iot-ubuntucore-20.pxu) / [camera-automated](https://github.com/canonical/checkbox/blob/main/providers/base/units/camera/test-plan.pxu#L89)  and may require refactoring of the logic in [camera_test.py](https://github.com/canonical/checkbox/blob/main/providers/base/bin/camera_test.py).

Based on [this part](https://github.com/canonical/checkbox/blob/main/providers/base/bin/camera_test.py#L577) of the code, if resolutions is provided, Clutter and Gtk will be initialized. However, both of them require a GUI for initialization. And that causing the job to fail.

     if args.test in ['display', 'still', 'led', 'resolutions']:
        import gi
        gi.require_version('Gst', '1.0')
        from gi.repository import Gst
        gi.require_version('GLib', '2.0')
        from gi.repository import GLib
        gi.require_version('Clutter', '1.0')
        from gi.repository import Clutter
        gi.require_version('Gtk', '3.0')
        from gi.repository import Gtk
        Gst.init(None)
        Clutter.init()
        Gtk.init([])

However, we tried to work around the issue by running the function without initializing Clutter and Gtk, and it seems to work fine.
https://certification.canonical.com/hardware/202305-31566/submission/323229/test/200875/result/34817873/ 

### To Reproduce

1. A machine support camera with a camera attached.
2. Run job `camera/multiple-resolution-images.*` on server or core
3. Job will fail while initial clutter

### Environment

- OS: Ubuntu 22.04.2 LTS
- Checkbox Type: Snap
- Checkbox Version: checkbox22	2.8 rev 437
- Hardware being tested: [devices](https://certification.canonical.com/hardware/202305-31566/submission/323043/devices/#/)

### Relevant log output

```shell
 (camera_test.py:15926): Clutter-CRITICAL **: 07:01:18.542: Unable to initialize Clutter: Could not initialize Gdk
 (camera_test.py:15926): Gtk-WARNING **: 07:01:18.543: cannot open display:
```
https://certification.canonical.com/hardware/202305-31566/submission/323043/test/76060/result/34787566/

**Comment by pieqq** [comment]
The things to take into account are:

- the script should work with desktop images (with a GUI) and with server/core images (not Gtk available)
- MIPI camera don't seem to work with `fswebcam` (see #264 )

It should be possible to use gstreamer without Gtk, that is to perform the image capture (and save it to file) without using Gtk, but only Gst.

**Comment by pieqq** [comment]
Todo:

1. Investigate what parts (`detect,led,display,still,resolutions`) really require Gtk/Clutter to run
2. Isolate the use of Gtk/Clutter to these
3. [Add some tests](https://github.com/canonical/checkbox/blob/main/CONTRIBUTING.md#writing-and-running-unit-tests-for-checkbox)
4. Test the changes on 22.04 classic, 22.04 server and UC22 (using side-loaded provider should be enough)

If done well, it might be possible to remove `fswebcam` from the script entirely.

**Comment by zongminl** [comment]
Fixed by #1400
